subsys/bluetooth/shell/cap_acceptor.c

```diff
@@ -261,3 +262,82 @@ SHELL_STATIC_SUBCMD_SET_CREATE(cap_acceptor_cmds,
 
 SHELL_CMD_ARG_REGISTER(cap_acceptor, &cap_acceptor_cmds, "Bluetooth CAP acceptor shell commands",
 		       cmd_cap_acceptor, 1, 1);
+
+ssize_t cap_acceptor_ad_data_add(struct bt_data *data_array,
+				 size_t data_array_size,
+				 bool discoverable)
+{
+	size_t ad_len = 0;
+
+	if (!discoverable) {
+		return ad_len;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_CAP_ACCEPTOR_SET_MEMBER) && cap_csip_svc_inst != NULL) {
+		static const uint8_t ad_cap_announcement[3] = {
+			BT_UUID_16_ENCODE(BT_UUID_CAS_VAL),
+			BT_AUDIO_UNICAST_ANNOUNCEMENT_TARGETED,
+		};
+		static uint8_t ad_rsi[BT_CSIP_RSI_SIZE];
+		int err;
+
+		__ASSERT(data_array_size > ad_len, "No space for AD_CAP_ANNOUNCEMENT");
+		data_array[ad_len].type = BT_DATA_SVC_DATA16;
+		data_array[ad_len].data_len = ARRAY_SIZE(ad_cap_announcement);
+		data_array[ad_len].data = &ad_cap_announcement[0];
+		ad_len++;
```

Comment:
any reason `cap_acceptor_ad_data_add()` cannot be reused here?

Reply:
Added a new function, `ad_cap_announcement_data_add`, that adds this part, and then use them in the two implementations of `cap_acceptor_ad_data_add`